osv/types.go

```diff
@@ -0,0 +1,48 @@
+package osv
+
+type OsvAffected struct {
+	Package  *OsvPackage `json:"package,omitempty"`
+	Ranges   []OsvRange  `json:"ranges,omitempty"`
+	Versions []string    `json:"versions,omitempty"`
+	// Ecosystem *OsvEcosystem `json:"ecosystem_specific,omitempty"` //The meaning of the values within the object is entirely defined by the ecosystem
+	// Database  *OsvDatabase  `json:"database_specific,omitempty"` //The meaning of the values within the object is entirely defined by the database
+}
+type OsvPackage struct {
+	Ecosystem string `json:"ecosystem,omitempty"`
+	Name      string `json:"name,omitempty"`
+	Purl      string `json:"purl,omitempty"`
+}
+type OsvRange struct {
+	Type   string     `json:"type,omitempty"`
+	Repo   string     `json:"repo,omitempty"`
+	Events []OsvEvent `json:"events,omitempty"`
+}
+type OsvEvent struct {
+	Introduced string `json:"introduced,omitempty"`
+	Fixed      string `json:"fixed,omitempty"`
+}
+
+// type OsvEcosystem struct {
+// }
+
+// type OsvDatabase struct {
+// }
+
+type OsvReference struct {
+	Type string `json:"type,omitempty"`
+	Url  string `json:"url,omitempty"`
+}
+
+//collection based on https://ossf.github.io/osv-schema/
+type OsvJson struct {
```

Comment:
`Json` postfix isn't very good choice as structure can be used some way which is not related to json serialization

Reply:
changed